src/pipelinedefinition.cpp

```diff
@@ -384,119 +387,149 @@ class NodeValidator {
+        // Affect shape by demultiplexer/gather if applies.
+        const auto& tensorInput = this->inputsInfo.at(modelInputName);
+        const auto& tensorOutput = this->dependencyOutputsInfo.at(modelOutputName);
+        shape_t tensorInputShape = tensorInput->getShape();
+        shape_t tensorOutputShape = tensorOutput->getShape();
+        if (dependencyNodeInfo.demultiplyCount) {
+            auto result = influenceShapeWithDemultiplexer(tensorOutputShape, dependencyNodeInfo);
+            if (!result.ok()) {
+                return result;
+            }
+        }
+        if (dependantNodeInfo.gatherFromNode.size() == 1) {
+            std::vector<NodeInfo>::const_iterator demultiplicatorNode;
+            auto result = getDependencyNodeInfo(*dependantNodeInfo.gatherFromNode.begin(), demultiplicatorNode);
+            if (!result.ok()) {
+                return result;
+            }
+            result = influenceShapeWithDemultiplexer(tensorInputShape, *demultiplicatorNode);
+            if (!result.ok()) {
+                return result;
+            }
+        } else if (dependantNodeInfo.gatherFromNode.size() > 1) {
+            SPDLOG_LOGGER_ERROR(modelmanager_logger, "Validation of pipeline({}) definition failed. Manual gathering from multiple nodes is not supported in node name: {}",
+                this->pipelineName,
+                dependantNodeInfo.nodeName);
+            return StatusCode::PIPELINE_MANUAL_GATHERING_FROM_MULTIPLE_NODES_NOT_SUPPORTED;
+        }
+        if (tensorInputShape != tensorOutputShape) {
+            SPDLOG_LOGGER_ERROR(modelmanager_logger, "Validation of pipeline({}) definition failed. Shape mismatch between: dependant node:{}; input:{}; shape:{} vs dependency node:{}; output:{}; shape:{}",
                 pipelineName,
                 dependantNodeInfo.nodeName,
-                dependantNodeInfo.modelName,
-                dependantNodeInfo.modelVersion.value_or(0),
                 modelInputName,
                 TensorInfo::shapeToString(tensorInput->getShape()),
                 dependencyNodeInfo.nodeName,
-                dependencyNodeInfo.modelName,
-                dependencyNodeInfo.modelVersion.value_or(0),
                 modelOutputName,
                 TensorInfo::shapeToString(tensorOutput->getShape()));
             return StatusCode::INVALID_SHAPE;
         }
         if (tensorInput->getPrecision() != tensorOutput->getPrecision()) {
-            SPDLOG_LOGGER_ERROR(modelmanager_logger, "Validation of pipeline({}) definition failed. Precision mismatch between: dependant node:{}; model:{}; version:{}; input:{}; precision:{} vs dependency node:{}; model:{}; version:{}; output:{}; precision:{}",
+            SPDLOG_LOGGER_ERROR(modelmanager_logger, "Validation of pipeline({}) definition failed. Precision mismatch between: dependant node:{}; input:{}; precision:{} vs dependency node:{}; output:{}; precision:{}",
```

Comment:
The same as in L451: `Validation of pipeline: {}`

Reply:
Done